### ISSUE TITLE
No camera support - video isn't supported in this browser yet.

### ISSUE THREAD

**Issue by leifrod** [wavebox/waveboxapp#1107]
* Wavebox Version: 4.11.3
* Operating System & Version: Ubuntu 18.10
* (Linux only) Install method: Debian installer, Snap Package
* Account type (if applicable):Office 365
Hi
I have tested both deb and snap installs and no camera available.
This is something to do with user agent I guess?
I have tried some custom UAs but no luck
Does anyone know a workaround?
I really need Teams calling.

Best wishes,

Greg

**Comment by nmat** [comment]
I have been testing out teams quite a bit. But from my experience in order to do calls, videos etc through teams you would need the desktop electron app.

Now this does however not mean that some functionality does not work. For example in my experience I have been able to do voice calls and be in voice calls using wavebox, browser etc. I have had problems making calls out but can receive calls. 

To be fair. The first part in this would be first to look if browser (cloud) service support the functionality and then have it be implemented as support in wavebox.

For example:
* Share screen (which is part of the desktop client)
This at the moment does not work neither in chrome or in wavebox. Since the cloud based service does not support it yet.

So best way to test this is to check if:
1. Is the service working in a browser (chrome is suggested to test in)
1.1 If it is supported then it can most likely be implemented if there is html5/javascript based support for the service

Can you make video calls using browser or does the button say "this will be implemented soon" or something?

**Comment by Thomas101** [comment]
I've just given this a try here and it seems to work. Could you try the following - just in case it highlights any other issue or something that I'm doing differently...

1. Check that access to the camera is available
    1. Add a new account and use the "Add any Website" button.
    2. Paste `https://davidwalsh.name/demo/camera.php` as the URL
    3. Does the webcam stream open up there?
2. Try a fresh add of Microsoft teams
    1. Add a new account and pick `Microsoft Teams` from the list
    2. Sign in and click the following button at the bottom of a teams conversation...
![Screenshot 2019-09-18 at 09 24 30](https://user-images.githubusercontent.com/103586/65131198-b8f8b500-d9f6-11e9-96fd-c64cefdae7ff.png)
    3. Does the video preview open up here?


I've just tried this with the Snap install and it seemed to work. Would be great to know where t differs your end

**Comment by leifrod** [comment]
Hi
I have tried the camera link and it works fine.
I have added Teams separately and allowed it access to camera (there was a popup) but the button you have I do not. After granting the camera permission via popup I have Video isn't supported in your browser yet on hover over the top right camera icon.
So it is the user agent that is the problem.
I tried both snap and deb installs, no luck.

**Comment by Thomas101** [comment]
Have you changed any of the UserAgent settings in Wavebox? There are a couple of places that would affect it...

1. Under `Settings` > `Advanced` have you disabled `Automatically Polyfill UserAgents`? (Ideally this should be enabled)
![Screenshot 2019-09-19 at 12 41 15](https://user-images.githubusercontent.com/103586/65241177-d8690e00-dada-11e9-9b8a-279b159dcad4.png)
2. Under `Account Settings` > `Advanced` have you enabled & set `Use custom UserAgent`? (Ideally this should be disabled and there should be no UserAgent set)
![Screenshot 2019-09-19 at 12 41 06](https://user-images.githubusercontent.com/103586/65241249-fe8eae00-dada-11e9-80ba-0de9721d28ac.png)

**Comment by leifrod** [comment]
Hi

Settings are default. I tried everything. (Default user agent, various ones 
reported workung etc)Does not work.
Grzegorz Szyło

**Comment by nmat** [comment]
I did some testing also using below information:

OS: Windows 10 1903 (OS Build 18362.356)
Wavebox version: 4.11.3
Wire config version: 2.1.1

Microsoft teams - Video meeting + voice testing and external/internal camera

**External Camera:** Logitech

**Summary:**
All working as intended, sound was perfect. Both external and internal camera worked as intended.

Sound on the other end worked perfectly also. No problems at all doing the call. This was as mentioned tested using both internal and external tools like mic, camera and speakers.

**Suggestion:**
When you have microsoft teams in wavebox active. Can you check the settings for the camera in the teams connected service? Are they set correctly?

**Comment by nmat** [comment]
Further note!

I can see that on individuals the camera icon is saying that the "Video isn't supported in your browser yet" but not in group or team conferencing.

So it seems like you can't make a one on one video call only group video call.

**Comment by Thomas101** [comment]
Thanks @nmat now I see the same. As you mentioned video calling works in a group, but not one-to-one. I've tried the same in Chrome and in Edge and they both exhibit the same behaviour.

I wonder why they've disabled one and not the other :-/

**Comment by leifrod** [comment]
Hi

You mentioned Teams camera settings in wavebox? I can't find any and just 
to clarify this is a Ubuntu 18 deb install, not Windows.
I tried group calling and yes it works bit not on individual chats. Weird
Grzegorz Szyło

**Comment by nmat** [comment]
@leifrod yes it seems like even if you are not using wavebox, that is if you use a browser like chrome the one-to-one does not work. So its more about what microsoft implements support for.

I have looked around a bit but can't find a reason for the difference of group chat vs one-to-one calls.

So the message is correct. Microsoft Teams for browser does not yet support one-to-one video calls.

You can always test the diff in a normal browser to see if you get the same result.

**Comment by nmat** [comment]
https://docs.microsoft.com/en-us/MicrosoftTeams/get-clients#web-client

This seems like the correct place to look for the support of various items in microsoft teams.

**Comment by nmat** [comment]
@leifrod can we maybe close this issue? Because it seems like this has more to do with what MS supports and do not support using web-client. Because it did work in group chat correct?
